src/include/parser/postgresparser.h

```diff
@@ -78,106 +95,105 @@ class PostgresParser {
+  static std::unique_ptr<TableRef> FromTransform(ParseResult *parse_result, SelectStmt *select_root);
+  static std::unique_ptr<GroupByDescription> GroupByTransform(ParseResult *parse_result, List *group,
+                                                              Node *having_node);
+  static std::unique_ptr<OrderByDescription> OrderByTransform(ParseResult *parse_result, List *order);
+  static common::ManagedPointer<AbstractExpression> WhereTransform(ParseResult *parse_result, Node *root);
 
   // FromTransform helpers
-  static std::unique_ptr<JoinDefinition> JoinTransform(JoinExpr *root);
+  static std::unique_ptr<JoinDefinition> JoinTransform(ParseResult *parse_result, JoinExpr *root);
   static std::string AliasTransform(Alias *root);
-  static std::unique_ptr<TableRef> RangeVarTransform(RangeVar *root);
-  static std::unique_ptr<TableRef> RangeSubselectTransform(RangeSubselect *root);
+  static std::unique_ptr<TableRef> RangeVarTransform(ParseResult *parse_result, RangeVar *root);
+  static std::unique_ptr<TableRef> RangeSubselectTransform(ParseResult *parse_result, RangeSubselect *root);
 
   // COPY statements
-  static std::unique_ptr<CopyStatement> CopyTransform(CopyStmt *root);
+  static std::unique_ptr<CopyStatement> CopyTransform(ParseResult *parse_result, CopyStmt *root);
 
   // CREATE statements
-  static std::unique_ptr<SQLStatement> CreateTransform(CreateStmt *root);
-  static std::unique_ptr<SQLStatement> CreateDatabaseTransform(CreateDatabaseStmt *root);
-  static std::unique_ptr<SQLStatement> CreateFunctionTransform(CreateFunctionStmt *root);
-  static std::unique_ptr<SQLStatement> CreateIndexTransform(IndexStmt *root);
-  static std::unique_ptr<SQLStatement> CreateSchemaTransform(CreateSchemaStmt *root);
-  static std::unique_ptr<SQLStatement> CreateTriggerTransform(CreateTrigStmt *root);
-  static std::unique_ptr<SQLStatement> CreateViewTransform(ViewStmt *root);
+  static std::unique_ptr<SQLStatement> CreateTransform(ParseResult *parse_result, CreateStmt *root);
+  static std::unique_ptr<SQLStatement> CreateDatabaseTransform(ParseResult *parse_result, CreateDatabaseStmt *root);
+  static std::unique_ptr<SQLStatement> CreateFunctionTransform(ParseResult *parse_result, CreateFunctionStmt *root);
+  static std::unique_ptr<SQLStatement> CreateIndexTransform(ParseResult *parse_result, IndexStmt *root);
+  static std::unique_ptr<SQLStatement> CreateSchemaTransform(ParseResult *parse_result, CreateSchemaStmt *root);
+  static std::unique_ptr<SQLStatement> CreateTriggerTransform(ParseResult *parse_result, CreateTrigStmt *root);
+  static std::unique_ptr<SQLStatement> CreateViewTransform(ParseResult *parse_result, ViewStmt *root);
 
   // CREATE helpers
   using ColumnDefTransResult = struct {
     std::unique_ptr<ColumnDefinition> col_;
-    std::vector<std::shared_ptr<ColumnDefinition>> fks_;
+    std::vector<std::unique_ptr<ColumnDefinition>> fks_;
```

Comment:
nit: fks_ doesn't seem like a very description name. I'm not sure what this is

Reply:
Foreign keys, I will add a comment.